client/src/components/Footwear.tsx

```diff
@@ -1,32 +1,50 @@
 import { Link } from 'react-router-dom';
+import { useContext } from 'react';
+import { closeContext } from './NavDrawerCloseContext';
 
 type Footwear = {
   isOpen: boolean;
+  handleClose: () => void;
 };
 
-export function Footwear({ isOpen }: Footwear) {
+export function Footwear({ isOpen, handleClose }: Footwear) {
+  const { closeNavDrawer } = useContext(closeContext);
+
   return (
     <>
       {isOpen && (
         <div>
-          <ul className="ml-3 cursor-pointer">
+          <ul className="ml-3 cursor-pointer" onClick={closeNavDrawer}>
```

Comment:
I think this should be `handleClose`. Then you won't need `closeNavDrawer` here anymore.